internal/discoverer/zookeeper_test.go

```diff
@@ -0,0 +1,109 @@
+package discoverer
+
+import (
+	"testing"
+
+	"github.com/api7/apisix-seed/internal/conf"
+	"github.com/api7/apisix-seed/internal/core/comm"
+	"github.com/api7/apisix-seed/internal/utils"
+	"github.com/go-zookeeper/zk"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v3"
+)
+
+var zkName = "zookeeper"
+var zkPrefix = "/zookeeper"
+var zkService = "svc"
+var zkNode1 = "{\"host\":\"127.0.0.1:1980\",\"weight\":100}"
+var zkNode2 = "{\"host\":\"127.0.0.1:1981\",\"weight\":100}"
+var zkYamlConfig = `
+hosts:
+  - "127.0.0.1:2181"
+prefix: /zookeeper
+weight: 100
+timeout: 10
+`
+
+func GetZkConfig(t *testing.T) *conf.Zookeeper {
+	zkConf := &conf.Zookeeper{}
+	err := yaml.Unmarshal([]byte(zkYamlConfig), zkConf)
+	assert.Nil(t, err)
+
+	return zkConf
+}
+
+func updateZkService(t *testing.T, conn *zk.Conn) {
+	svcPath := zkPrefix + "/" + zkService
+	_, stat, err := conn.Get(svcPath)
+	assert.Nil(t, err)
+	_, err = conn.Set(svcPath, []byte(zkNode2), stat.Version)
+	assert.Nil(t, err)
+}
+
+func createZkService(t *testing.T, conn *zk.Conn) {
+	_, err := conn.Create(zkPrefix+"/"+zkService, []byte(zkNode1), 0, zk.WorldACL(zk.PermAll))
+	assert.Nil(t, err)
+}
+
+func removeZkService(t *testing.T, conn *zk.Conn) {
+	svcPath := zkPrefix + "/" + zkService
+	_, stat, err := conn.Get(svcPath)
+	if err != nil {
+		return
+	}
+	err = conn.Delete(zkPrefix+"/"+zkService, stat.Version)
+	assert.Nil(t, err)
+}
+
+func TestNewZkDiscoverer(t *testing.T) {
+	config := GetZkConfig(t)
+	assert.NotNil(t, config)
+	dis, err := NewZookeeperDiscoverer(config)
+	assert.Nil(t, err)
+	assert.NotNil(t, dis)
+}
+
+func TestZkDiscoverer(t *testing.T) {
+	config := GetZkConfig(t)
+	assert.NotNil(t, config)
+	dis, err := Discoveries[zkName](config)
+	conn := dis.(*ZookeeperDiscoverer).zkConn
+	removeZkService(t, conn)
+	assert.Nil(t, err)
+	headers := []string{utils.EventAdd, "/apisix/routes/1", zkService}
+	query, err := comm.NewQuery(headers, map[string]string{})
+	assert.Nil(t, err)
+	err = dis.Query(&query)
+	assert.NotNil(t, err)
+	msg := dis.Watch()
+	createZkService(t, conn)
+	zkDiscovererRegister(t, msg)
+	updateZkService(t, conn)
+	zkDiscovererUpdate(t, msg)
+	removeZkService(t, conn)
+	zkDiscovererRemove(t, msg)
+}
```

Comment:
```golang
func createZkService(node string) error {
		conn := initZkConn(t)
	_, err := conn.Create(zkPrefix+"/"+zkService, []byte(node), 0, zk.WorldACL(zk.PermAll))

	return err
}

func getNode(msg chan *comm.Message) string {
	m := <-msg
	_, _, nodes, _ := m.Decode()
	
	return node[0]
}

func TestZkDiscoverer(t *testing.T) {
	...
	msg := dis.Watch()
	err := createZkService("{\"host\":\"127.0.0.1:1980\",\"weight\":100}")
	assert.Nil(t, err)
	assert.Equal(t, getNode(msg), "127.0.0.1:1980")
}
```
I suggest that your unit tests can be written like this. This can more intuitively reflect the purpose of your unit test. The input and output are together, which is very intuitive. All assertion judgments are also centralized rather than distributed in various utility functions